From: David Cohen <david.a.cohen@intel.com>
Subject: [PATCH] usb: android: make function configuration more verbose

We need more kernel verbosity WRT userspace's usb gadget function
configuration, as the kernel behavior is highly dependent of it.
This patch will add extra printks to improve USB OTG debugging. But these
printks are one time used per USB OTG configuration. Since it happens seldom,
we are not flooding console.

Signed-off-by: David Cohen <david.a.cohen@intel.com>

---
 drivers/usb/gadget/android.c |    4 ++++
 1 file changed, 4 insertions(+)

--- linux.orig/drivers/usb/gadget/android.c
+++ linux/drivers/usb/gadget/android.c
@@ -1208,6 +1208,8 @@ functions_store(struct device *pdev, str
 	strlcpy(buf, buff, sizeof(buf));
 	b = strim(buf);
 
+	pr_info("android_usb: selecting functions: %s\n", b);
+	
 	while (b) {
 		name = strsep(&b, ",");
 		if (!name)
@@ -1283,6 +1285,7 @@ static ssize_t enable_store(struct devic
 		cdev->desc.bDeviceSubClass = device_desc.bDeviceSubClass;
 		cdev->desc.bDeviceProtocol = device_desc.bDeviceProtocol;
 		list_for_each_entry(f, &dev->enabled_functions, enabled_list) {
+			pr_info("android_usb: enabling function %s\n", f->name);
 			if (f->enable)
 				f->enable(f);
 		}
@@ -1291,6 +1294,7 @@ static ssize_t enable_store(struct devic
 	} else if (!enabled && dev->enabled) {
 		android_disable(dev);
 		list_for_each_entry(f, &dev->enabled_functions, enabled_list) {
+			pr_info("android_usb: disabling function %s\n", f->name);
 			if (f->disable)
 				f->disable(f);
 		}
